fem/tmop_tools.hpp

```diff
@@ -245,14 +265,40 @@ class TMOPNewtonSolver : public LBFGSSolver
    {
       surf_fit_rel_change_threshold = threshold;
    }
+   /// Used for itaration-based surface fitting termination.
    void SetMaxNumberofIncrementsForAdaptiveFitting(int count)
    {
       max_adapt_inc_count = count;
    }
+   /// Used for error-based surface fitting termination.
    void SetTerminationWithMaxSurfaceFittingError(real_t max_error)
    {
       surf_fit_max_threshold = max_error;
+      surf_fit_converge_error = true;
+   }
```

Comment:
I think it will be simpler if there's a single function that does all these:
```
SetTerminationWithMaxSurfaceFittingError(real_t max_error)
SetMaxSurfaceFittingError(real_t max_error)
SetFittingConvergenceBasedOnError(bool mode)
```